// @react-login-page/page2/cjs/control/Email.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Email = void 0;
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _reactLoginPage = require("react-login-page");
var _email = require("../icons/email");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["keyname", "name", "rename"];
var Email = function Email(props) {
  var _props$keyname = props.keyname,
    keyname = _props$keyname === void 0 ? 'email' : _props$keyname,
    name = props.name,
    rename = props.rename,
    elmProps = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var nameBase = name || rename || keyname;
  var key = keyname || name;
  if (!elmProps.children) {
    elmProps.children = _email.EmailIcon;
  }
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactLoginPage.Input, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    placeholder: "Email",
    spellCheck: false
  }, elmProps), {}, {
    type: "email",
    name: nameBase,
    keyname: key
  }));
};
exports.Email = Email;
Email.displayName = 'Login.Email';
